src/adapter/src/coord.rs

```diff
@@ -3782,6 +3782,11 @@ pub fn serve(
             )
             .await;
         }
+
+        // Opening the durable catalog uses one or more timestamps without communicating with
+        // the timestamp oracle. Here we make sure to apply the catalog upper with the timestamp
+        // oracle to linearize future operations with opening the catalog.
```

Comment:
This is the exception to the "don't apply catalog timestamps" rule. This happens at the end of startup and doesn't have a negative impact on read latency because we're going to do a bunch of table writes after this which bumps and applies the oracle timestamp.